### ISSUE TITLE
Effect of mean time to death...and general checking understanding

### ISSUE THREAD

**Issue by birdsarah** [jwrichar/COVID19-mortality#6]
Hi!

My colleage @ilanasegall and I just took a real deep dive to try and understand the US states case count model. 

Caveat: She's a statistician but hasn't used pymc3. I'm a research engineer with engineering background and ramping up on statistics. Neither of us have a public health background.

Our basic understanding can be summarized as follows:
* mu_0 is the nationwide mortality rate
* the binomial likelihood gets us the p(observed_deaths=d|p=mu)
* the binomial likelihood is parameterized by the national mortality rate mu_0 and c_star (the estimated number of cases *from 7 days ago*)
* c_star has to be estimated as we know testing isn't saturated, so there is a logistic regression with the variable x being the tests per population.

It took us a long time to figure this out because we're use to "n" being something you know not something you have to estimate from another function.

Small thing 1 : We got confused because [`mu_model_logit`](https://github.com/jwrichar/COVID19-mortality/blob/master/source/model.py#L116) is the mu of "n" (feeding into c_star). Rather than the `mu_i`s of different mortality rates.

Small thing 2:  We got confused about the number of observed deaths being the smaller of the deaths today, or the number of positive cases 7 days ago. Texas, for example, had 39 cases 7 days ago, but 5 deaths today. For states that have had no deaths (which my understanding of when this would kick in) why is the 7 day ago rate better.

After this basic understanding, our understanding gets a little fuzzy as to why this all works and the relationship between deaths today, number of cases positive 7 days ago, and number of tests conducted 7 days ago.

We rewrote the model code as follows to help get some clarity:

```
def case_count_model_us_states(df):
    n_states = len(df)

    # For each state, let:
    
    # c_obs = number of observed cases 7 days ago
    c_obs_7_days_ago = df['num_pos_7_days_ago'].values
    # d_obs = number of observed deaths (used to be a min with case_obs_7_days_back)
    d_obs_today = df.death.values
    # people per test: normalize to South Korea
    # assuming S.K. testing is "saturated"
    ppt_sk = np.log10(51500000. / 250000)
    people_per_test_7_days_ago = (np.log10(df.people_per_test_7_days_ago) - ppt_sk).values

    covid_case_count_model = pm.Model()
    with covid_case_count_model:
        # Priors:
        mu_0 = pm.Beta('mu_0', alpha=1, beta=100, testval=0.01)
        b_0 = pm.Bound(pm.Normal, lower=0.0)('b_0', mu=8, sigma=3, shape=1)
        b_1 = pm.Bound(pm.Normal, upper=0.0)('b_1', mu=-1, sigma=1, shape=1)
        sigma = pm.HalfNormal('sigma', sigma=0.5, testval=0.1)
        
        # Model probability of case under-reporting as logistic regression:
        mu_model_logit = b_0 + b_1 * people_per_test_7_days_ago
        tau_logit = pm.Normal('tau_logit', mu=mu_model_logit, sigma=sigma, shape=n_states)
        tau = np.exp(tau_logit) / (np.exp(tau_logit) + 1)

        c_star_7_days_ago = c_obs_7_days_ago / tau

        # Binomial likelihood:
        d = pm.Binomial('d', n=c_star_7_days_ago, p=mu_0, observed=d_obs_today)

    return covid_case_count_model
```

So we're left with two big questions that we're hoping we can understand better:

1. Why model deaths against the adjusted number of positive cases from 7 days ago. We thought the mean time to death was much longer - like 18 days. (We know that it may just be because the data last week was terrible - if that's the case, presumably we can expect these estimates to worsen as we have the data to increase the time between positive cases n days ago and deaths today).
2. Why not use `fillna(0)` on deaths and just take the deaths as the deaths?

**Comment by jwrichar** [comment]
@birdsarah  @ilanasegall -- thanks for taking the time to go though the model code so thoroughly.  And apologies for the messy code / notation!

Your understanding of the current model is spot on.  There clearly are a number of built-in assumptions here, but the fundamental one that ties everything together is that **the underlying mortality rate should be the same across all states** (perhaps subject to covariates like average age of population, hospital beds per capita, etc. -- things that I'd love to add later!)  And right now test volume is clearly the main driver in the observed disparate mortality rates across different states.

So that brings us to your (big question 1) - why 7 days?  Honestly, convenience!  When i first built this, the data on day T-7 were absolute shit.  But, the more I've thought about it, 7 days isn't a terrible assumption because (a) it takes a couple days for test results to come back and (b) people aren't necessarily getting tested at the onset of symptoms (and those symptoms can appear days after getting the virus).  At any rate, I think there's an opportunity to tweak the 7-day window as more information comes in about the virus -- and it actually *could* be treated as an unknown parameter :)

For the (big question 2) - when I first built this, LA had 4 deaths and only 1 case a week prior.  So clearly could not work in the model (the Binomial nan'd out).

At any rate, thanks for re-writing the model code -- it's much clearer now.  Feel free to open up a PR for that.

Let me know if you have any follow-up questions.  I realize there's a number of built-in assumptions here, so maybe I'll take a stab at enumerating them and also chronicling all of the other (failed!) model attempts.

**Comment by birdsarah** [comment]
Thanks for the quick reply.

> fundamental one that ties everything together is that the underlying mortality rate should be the same across all states 

Thinking about it more, I thought that the assumption is the ratio between testing rate and mortality is the same across states.

Which made me realize that the 7 days doesn't matter. If I'm right it doesn't matter that the mean time to death isn't 7 days the model is constructing a mortality rate based on number of deaths 7 days after a certain number of positive tests. So you could just as easily do 3 or 4 days - it would just be a different, lower, rate.

With NY now getting overwhelmed, my husband told me they're now only testing people for whom a test would change the state of care they're receiving, I'm thinking it would be good to remove them from the model.  Similarly WA is a special case where the mortality to testing ratio is off because they were at the front line. It may be better to model based on results for the remaining states that aren't in such a crisis....thoughts?

**Comment by waldren** [comment]
Thanks for putting this repo together. On the public health/ medicine side, There was an article published on spread by undocumented (no test/Sx) published in Science: https://science.sciencemag.org/content/early/2020/03/13/science.abb3221  There is a good summary by a doc doing a daily youtube: https://www.youtube.com/watch?v=AToF8O5T86s&t=204s 

Big Kudos!

**Comment by jwrichar** [comment]
> Thinking about it more, I thought that the assumption is the ratio between testing rate and mortality is the same across states.

Yeah, that's a big (unstated) assumption as well -- good shout-out.  I'd love to add population density to that logistic regression function in the off chance that pop dens also factors into mortality rates.  Also avg age of population and (critically!) number of ventilators per capita.

> Which made me realize that the 7 days doesn't matter. If I'm right it doesn't matter that the mean time to death isn't 7 days the model is constructing a mortality rate based on number of deaths 7 days after a certain number of positive tests. So you could just as easily do 3 or 4 days - it would just be a different, lower, rate.

Yeah, that makes sense.  Unless we're trying to explicitly estimate mortality rate (digression: which was actually one of my initial aims, but I decided to take it out of the dashboard because it was confusing to people + I wasn't super confident in the result), then the time delay actually doesn't matter too much (at least to first order) since it shouldn't really affect the testing-vs-reporting curve (maybe??).

> With NY now getting overwhelmed, my husband told me they're now only testing people for whom a test would change the state of care they're receiving, I'm thinking it would be good to remove them from the model. Similarly WA is a special case where the mortality to testing ratio is off because they were at the front line. It may be better to model based on results for the remaining states that aren't in such a crisis....thoughts?

Yeah, that's a good idea.  Generally it's a good sanity check to see if the model converges to a similar result without some of the data (there's a whole suite of "sensitivity analysis" tools/approaches that aim to do this).  The only slight issue I see in removing NY/WA is that those are the two states with the highest testing volume (# people / test) in the US, so estimates for that part of the curve might degrade.  Either way, it's a good idea to test!

**Comment by jwrichar** [comment]
@waldren thanks for the resources!  That paper is very interesting, and frightening!  The more I read, the more it becomes clear that the % of asymptomatic cases is much higher than anyone first believed.

Stay safe, everybody.
